### PR TITLE
feat(AutoMod): add support for custom_message

### PR DESCRIPTION
**Pull request by Victorsitou**
## Summary

https://github.com/discord/discord-api-docs/pull/5955

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/automod.py [hunk 1]
```diff
@@ -127,11 +135,21 @@ class AutoModBlockMessageAction(AutoModAction):
 
     _metadata: AutoModBlockMessageActionMetadata
 
-    def __init__(self) -> None:
+    def __init__(self, custom_message: str) -> None:
```
Comment (onerandomusername) [inline disnake/automod.py:138]: this should have a default of None

File: disnake/automod.py [hunk 2]
```diff
@@ -116,6 +116,14 @@ class AutoModBlockMessageAction(AutoModAction):
 
     .. versionadded:: 2.6
 
+    Parameters
+    ----------
+    custom_message: Optional[:class:`str`]
+        The custom message to send to the user when the rule is triggered.
```
Comment (shiftinv) [inline disnake/automod.py:122]:
```suggestion
        The custom message to show to the user when the rule is triggered.
```

File: disnake/automod.py [hunk 3]
```diff
@@ -127,11 +135,22 @@ class AutoModBlockMessageAction(AutoModAction):
...
+        if custom_message is not None:
+            self._metadata["custom_message"] = custom_message
+
+    @property
+    def custom_message(self) -> str:
+        """:class:`str`: The custom message to send to the user when the rule is triggered.
+
+        .. versionadded:: 2.9
+        """
+        return self._metadata.get("custom_message", "")
```
Comment (shiftinv) [inline disnake/automod.py:150]:
```suggestion
    @property
    def custom_message(self) -> Optional[str]:
        """Optional[:class:`str`]: The custom message to show to the user when the rule is triggered.

        .. versionadded:: 2.9
        """
        return self._metadata.get("custom_message")
```
Better to make this optional, for clarity

File: disnake/types/automod.py [hunk 4]
```diff
@@ -15,6 +15,7 @@
 
 
 class AutoModBlockMessageActionMetadata(TypedDict):
+    custom_message: NotRequired[str]
     ...
```
Comment (shiftinv) [inline disnake/types/automod.py:19]:
```suggestion
```

**Review by shiftinv — COMMENTED** [review]
a couple nits, but looks good to me otherwise c:

File: disnake/automod.py (hunk 1 above)
Comment (Victorsitou) [reply disnake/automod.py]: 717deddb068512229a24b634bab8bbb729ce00ac

File: disnake/automod.py [hunk 5]
```diff
@@ -127,11 +135,21 @@ class AutoModBlockMessageAction(AutoModAction):
 
     _metadata: AutoModBlockMessageActionMetadata
 
-    def __init__(self) -> None:
+    def __init__(self, custom_message: Optional[str] = None) -> None:
         super().__init__(type=AutoModActionType.block_message)
 
+        self._metadata["custom_message"] = custom_message or ""
```
Comment (onerandomusername) [inline disnake/automod.py:141]: This should only be set if it is provided, right now this always provides custom_message to the API but it is optional.
```suggestion
        if custom_message is not None:
            self._metadata["custom_message"] = custom_message
```

**Review by onerandomusername — APPROVED** [review]

**Review by shiftinv — APPROVED** [review]
lgtm, thanks!

**Pull request merged**